File: shoring/desk/src/main/java/com/panopset/marin/oldswpkg/games/blackjack/BlackjackTable.java
package com.panopset.marin.oldswpkg.games.blackjack;

import com.panopset.compat.Tile;

public class BlackjackTable {

    private TableLayout tableLayout;
    private int w,h;

    private TableLayout getTableLayout() {
        if (tableLayout == null) {
            tableLayout = new TableLandscape(w,h);
        }
        return tableLayout;
    }

    public Tile getStatusTile() {
        return getTableLayout().getStatusTile();
    }

    public Tile getChipTray() {
        return getTableLayout().getChipTray();
    }

    public Tile getPlayerTile() {
        return getTableLayout().getPlayerTile();
    }

    public Tile getDealerTile() {
        return getTableLayout().getDealerTile();
    }

    public Tile getMsgTile() {
        return getTableLayout().getMsgTile();
    }

    public Tile getTableTile() {
        return getTableLayout().getTableTile();
    }

    private void setLandscape() {
        tableLayout = new TableLandscape(w,h);
    }

    private void setPortrait() {
        tableLayout = new TablePortrait(w,h);
    }

    public void setBounds(int width, int height) {
        if (width < 1 || height < 1) {
            throw new RuntimeException();
        }
        w = width;
        h = height;
        if (width > height) {
            setLandscape();
        } else {
            setPortrait();
        }
    }

    public void setCardHeight(int cardHeight) {
      getTableLayout().setCardHeight(cardHeight);
    }

}

File: code/shoring/desk/src/main/kotlin/com/panopset/marin/oldswpkg/games/blackjack/BlackjackTable.kt
package com.panopset.marin.oldswpkg.games.blackjack


class BlackjackTable(w: Int, h: Int, cardHeight: Int) {
    private var tableLayout: TableLayout
    = if (w > h) TableLandscape(w, h, cardHeight)  else TablePortrait(w, h, cardHeight)
    val statusTile = tableLayout.statusTile
    val chipTray = tableLayout.chipTray
    val playerTile = tableLayout.playerTile
    val dealerTile = tableLayout.dealerTile
    val msgTile = tableLayout.msgTile
}
